src/main/java/org/broadinstitute/consent/http/resources/DataSetResource.java

```diff
@@ -57,16 +58,28 @@
     private final String END_OF_LINE = System.lineSeparator();
     private final String TSV_DELIMITER = "\t";
     private final DataSetAPI api;
+    private final DataSetService dataSetService;
     private final DataAccessRequestAPI dataAccessRequestAPI;
     private final UserService userService;
 
     @Inject
-    public DataSetResource(UserService userService) {
+    public DataSetResource(DataSetService dataSetService, UserService userService) {
         this.dataAccessRequestAPI = AbstractDataAccessRequestAPI.getInstance();
         this.api = AbstractDataSetAPI.getInstance();
+        this.dataSetService = dataSetService;
         this.userService = userService;
     }
 
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/test")
+    @PermitAll
+    public Response createDataSet(@Context UriInfo info, DataSet dataSet) {
```

Comment:
One thing I want all of our API endpoints to do is to take `@Auth AuthUser` as an argument. That makes the currently authenticated user available to the endpoint. I would argue that we don't need the `UriInfo` in general. If we return a full object that has all of the required information in it, then the client can do with that what they wish.

Reply:
Additionally, for this method, we should be passing in a String instead and then deserializing that string inside the method. What this does is rely on Jackson to deserialize whatever object is passed in as a `DataSet`. This pattern is very difficult to debug and I find Jackson to be a huge pain and general security risk. For that reason, it should be standard practice for us to manage deserialization ourselves.